projects/timeswap/index.js

```diff
@@ -0,0 +1,78 @@
+const sdk = require("@defillama/sdk");
+
+const { getBlock } = require("../helper/getBlock");
+const { getChainTransform } = require("../helper/portedTokens");
+
+const factories = {
+  polygon: [
+    {
+      address: "0x37938633629B2D260c6dF2e6d62a7E8CfEe097c5",
+      startblock: 26192974,
+    },
+    {
+      address: "0x3a82fB3294a648368F5Ef07261Ec740B05Da9e85",
+      startblock: 26603572,
+    },
+  ],
+};
+
+function chainTvl(chain) {
+  return async (timestamp, _ethBlock, chainBlocks) => {
+    const block = await getBlock(timestamp, chain, chainBlocks);
+
+    const logs = [];
+
+    for (const factory of factories[chain]) {
+      logs.push(
+        ...(
+          await sdk.api.util.getLogs({
```

Comment:
@emmanuelantony2000 I think this is the issue, is there a way to get pairs some other way. querying events is painful and tends to break a lot

Reply:
@g1nt0ki apologies for the delayed response, we have set up a subgraph to query the pairs. We can use that.

Reply:
> @g1nt0ki apologies for the delayed response, we have set up a subgraph to query the pairs. We can use that.

Sounds great, let me know if you need any assistance with changing the adapter to use subgraph